File: packages/python/src/tesote_sdk/v3/_stubs.py
"""v3 resources not yet wired beyond their public shape."""

from __future__ import annotations

from typing import Any, Dict

from ..transport import Transport


class _StubResource:
    def __init__(self, transport: Transport) -> None:
        self._transport = transport


class TransactionsResource(_StubResource):
    def list_for_account(self, account_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 transactions.list_for_account not yet implemented")

    def get(self, transaction_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 transactions.get not yet implemented")

    def export(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v3 transactions.export not yet implemented")

    def sync(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v3 transactions.sync not yet implemented")

    def bulk(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v3 transactions.bulk not yet implemented")

    def search(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v3 transactions.search not yet implemented")


class SyncSessionsResource(_StubResource):
    def list(self, account_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 sync_sessions.list not yet implemented")

    def get(self, account_id: str, session_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 sync_sessions.get not yet implemented")


class TransactionOrdersResource(_StubResource):
    def list(self, account_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 transaction_orders.list not yet implemented")

    def get(self, account_id: str, order_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 transaction_orders.get not yet implemented")

    def create(self, account_id: str, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v3 transaction_orders.create not yet implemented")

    def submit(self, account_id: str, order_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 transaction_orders.submit not yet implemented")

    def cancel(self, account_id: str, order_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 transaction_orders.cancel not yet implemented")


class BatchesResource(_StubResource):
    def create(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v3 batches.create not yet implemented")

    def get(self, batch_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 batches.get not yet implemented")

    def approve(self, batch_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 batches.approve not yet implemented")

    def submit(self, batch_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 batches.submit not yet implemented")

    def cancel(self, batch_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 batches.cancel not yet implemented")


class PaymentMethodsResource(_StubResource):
    def list(self) -> Dict[str, Any]:
        raise NotImplementedError("v3 payment_methods.list not yet implemented")

    def get(self, payment_method_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 payment_methods.get not yet implemented")

    def create(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v3 payment_methods.create not yet implemented")

    def update(self, payment_method_id: str, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v3 payment_methods.update not yet implemented")

    def delete(self, payment_method_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 payment_methods.delete not yet implemented")


class CategoriesResource(_StubResource):
    def list(self) -> Dict[str, Any]:
        raise NotImplementedError("v3 categories.list not yet implemented")

    def get(self, category_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 categories.get not yet implemented")

    def create(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v3 categories.create not yet implemented")

    def update(self, category_id: str, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v3 categories.update not yet implemented")

    def delete(self, category_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 categories.delete not yet implemented")


class CounterpartiesResource(_StubResource):
    def list(self) -> Dict[str, Any]:
        raise NotImplementedError("v3 counterparties.list not yet implemented")

    def get(self, counterparty_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 counterparties.get not yet implemented")

    def create(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v3 counterparties.create not yet implemented")

    def update(self, counterparty_id: str, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v3 counterparties.update not yet implemented")

    def delete(self, counterparty_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 counterparties.delete not yet implemented")


class LegalEntitiesResource(_StubResource):
    def list(self) -> Dict[str, Any]:
        raise NotImplementedError("v3 legal_entities.list not yet implemented")

    def get(self, legal_entity_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 legal_entities.get not yet implemented")


class ConnectionsResource(_StubResource):
    def list(self) -> Dict[str, Any]:
        raise NotImplementedError("v3 connections.list not yet implemented")

    def get(self, connection_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 connections.get not yet implemented")

    def status(self, connection_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 connections.status not yet implemented")


class ReportsResource(_StubResource):
    def cash_flow(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v3 reports.cash_flow not yet implemented")


class BalanceHistoryResource(_StubResource):
    def list_for_account(self, account_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v3 balance_history.list_for_account not yet implemented")


class WorkspaceResource(_StubResource):
    def get(self) -> Dict[str, Any]:
        raise NotImplementedError("v3 workspace.get not yet implemented")


class McpResource(_StubResource):
    def handle(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v3 mcp.handle not yet implemented")


class StatusResource(_StubResource):
    def status(self) -> Dict[str, Any]:
        raise NotImplementedError("v3 status not yet implemented")

    def whoami(self) -> Dict[str, Any]:
        raise NotImplementedError("v3 whoami not yet implemented")


__all__ = [
    "TransactionsResource",
    "SyncSessionsResource",
    "TransactionOrdersResource",
    "BatchesResource",
    "PaymentMethodsResource",
    "CategoriesResource",
    "CounterpartiesResource",
    "LegalEntitiesResource",
    "ConnectionsResource",
    "ReportsResource",
    "BalanceHistoryResource",
    "WorkspaceResource",
    "McpResource",
    "StatusResource",
]

File: packages/ruby/lib/tesote_sdk/v3/stubs.rb
module TesoteSdk
  module V3
    # Resource clients documented in resources.md but not wired in 0.1.0.
    # Each method raises NotImplementedError so the surface is discoverable.

    class StubBase
      def initialize(transport)
        @transport = transport
      end
    end

    class Transactions < StubBase
      def list_for_account(_account_id, _query = {}, opts: {})
        raise NotImplementedError, 'V3::Transactions#list_for_account not wired in 0.1.0'
      end

      def get(_id, opts: {})
        raise NotImplementedError, 'V3::Transactions#get not wired in 0.1.0'
      end

      def export(_account_id, _params = {}, opts: {})
        raise NotImplementedError, 'V3::Transactions#export not wired in 0.1.0'
      end

      def sync(_account_id, opts: {})
        raise NotImplementedError, 'V3::Transactions#sync not wired in 0.1.0'
      end

      def bulk(_payload, opts: {})
        raise NotImplementedError, 'V3::Transactions#bulk not wired in 0.1.0'
      end

      def search(_query = {}, opts: {})
        raise NotImplementedError, 'V3::Transactions#search not wired in 0.1.0'
      end
    end

    class SyncSessions < StubBase
      def list(_account_id, _query = {}, opts: {})
        raise NotImplementedError, 'V3::SyncSessions#list not wired in 0.1.0'
      end

      def get(_account_id, _id, opts: {})
        raise NotImplementedError, 'V3::SyncSessions#get not wired in 0.1.0'
      end
    end

    class TransactionOrders < StubBase
      def list(_account_id, _query = {}, opts: {})
        raise NotImplementedError, 'V3::TransactionOrders#list not wired in 0.1.0'
      end

      def get(_account_id, _id, opts: {})
        raise NotImplementedError, 'V3::TransactionOrders#get not wired in 0.1.0'
      end

      def create(_account_id, _payload, opts: {})
        raise NotImplementedError, 'V3::TransactionOrders#create not wired in 0.1.0'
      end

      def submit(_account_id, _id, opts: {})
        raise NotImplementedError, 'V3::TransactionOrders#submit not wired in 0.1.0'
      end

      def cancel(_account_id, _id, opts: {})
        raise NotImplementedError, 'V3::TransactionOrders#cancel not wired in 0.1.0'
      end
    end

    class Batches < StubBase
      def create(_payload, opts: {})
        raise NotImplementedError, 'V3::Batches#create not wired in 0.1.0'
      end

      def get(_id, opts: {})
        raise NotImplementedError, 'V3::Batches#get not wired in 0.1.0'
      end

      def approve(_id, opts: {})
        raise NotImplementedError, 'V3::Batches#approve not wired in 0.1.0'
      end

      def submit(_id, opts: {})
        raise NotImplementedError, 'V3::Batches#submit not wired in 0.1.0'
      end

      def cancel(_id, opts: {})
        raise NotImplementedError, 'V3::Batches#cancel not wired in 0.1.0'
      end
    end

    class PaymentMethods < StubBase
      def list(_query = {}, opts: {})
        raise NotImplementedError, 'V3::PaymentMethods#list not wired in 0.1.0'
      end

      def get(_id, opts: {})
        raise NotImplementedError, 'V3::PaymentMethods#get not wired in 0.1.0'
      end

      def create(_payload, opts: {})
        raise NotImplementedError, 'V3::PaymentMethods#create not wired in 0.1.0'
      end

      def update(_id, _payload, opts: {})
        raise NotImplementedError, 'V3::PaymentMethods#update not wired in 0.1.0'
      end

      def delete(_id, opts: {})
        raise NotImplementedError, 'V3::PaymentMethods#delete not wired in 0.1.0'
      end
    end

    class Categories < StubBase
      def list(_query = {}, opts: {})
        raise NotImplementedError, 'V3::Categories#list not wired in 0.1.0'
      end

      def get(_id, opts: {})
        raise NotImplementedError, 'V3::Categories#get not wired in 0.1.0'
      end

      def create(_payload, opts: {})
        raise NotImplementedError, 'V3::Categories#create not wired in 0.1.0'
      end

      def update(_id, _payload, opts: {})
        raise NotImplementedError, 'V3::Categories#update not wired in 0.1.0'
      end

      def delete(_id, opts: {})
        raise NotImplementedError, 'V3::Categories#delete not wired in 0.1.0'
      end
    end

    class Counterparties < StubBase
      def list(_query = {}, opts: {})
        raise NotImplementedError, 'V3::Counterparties#list not wired in 0.1.0'
      end

      def get(_id, opts: {})
        raise NotImplementedError, 'V3::Counterparties#get not wired in 0.1.0'
      end

      def create(_payload, opts: {})
        raise NotImplementedError, 'V3::Counterparties#create not wired in 0.1.0'
      end

      def update(_id, _payload, opts: {})
        raise NotImplementedError, 'V3::Counterparties#update not wired in 0.1.0'
      end

      def delete(_id, opts: {})
        raise NotImplementedError, 'V3::Counterparties#delete not wired in 0.1.0'
      end
    end

    class LegalEntities < StubBase
      def list(_query = {}, opts: {})
        raise NotImplementedError, 'V3::LegalEntities#list not wired in 0.1.0'
      end

      def get(_id, opts: {})
        raise NotImplementedError, 'V3::LegalEntities#get not wired in 0.1.0'
      end
    end

    class Connections < StubBase
      def list(_query = {}, opts: {})
        raise NotImplementedError, 'V3::Connections#list not wired in 0.1.0'
      end

      def get(_id, opts: {})
        raise NotImplementedError, 'V3::Connections#get not wired in 0.1.0'
      end

      def status(_id, opts: {})
        raise NotImplementedError, 'V3::Connections#status not wired in 0.1.0'
      end
    end

    class Webhooks < StubBase
      def list(_query = {}, opts: {})
        raise NotImplementedError, 'V3::Webhooks#list not wired in 0.1.0'
      end

      def get(_id, opts: {})
        raise NotImplementedError, 'V3::Webhooks#get not wired in 0.1.0'
      end

      def create(_payload, opts: {})
        raise NotImplementedError, 'V3::Webhooks#create not wired in 0.1.0'
      end

      def update(_id, _payload, opts: {})
        raise NotImplementedError, 'V3::Webhooks#update not wired in 0.1.0'
      end

      def delete(_id, opts: {})
        raise NotImplementedError, 'V3::Webhooks#delete not wired in 0.1.0'
      end
    end

    class Reports < StubBase
      def cash_flow(_query = {}, opts: {})
        raise NotImplementedError, 'V3::Reports#cash_flow not wired in 0.1.0'
      end
    end

    class BalanceHistory < StubBase
      def list_for_account(_account_id, _query = {}, opts: {})
        raise NotImplementedError, 'V3::BalanceHistory#list_for_account not wired in 0.1.0'
      end
    end

    class Workspace < StubBase
      def get(opts: {})
        raise NotImplementedError, 'V3::Workspace#get not wired in 0.1.0'
      end
    end

    class Mcp < StubBase
      def handle(_payload, opts: {})
        raise NotImplementedError, 'V3::Mcp#handle not wired in 0.1.0'
      end
    end

    class Status < StubBase
      def status(opts: {})
        raise NotImplementedError, 'V3::Status#status not wired in 0.1.0'
      end

      def whoami(opts: {})
        raise NotImplementedError, 'V3::Status#whoami not wired in 0.1.0'
      end
    end
  end
end
